src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -244,6 +244,9 @@ public function setValue(&$objectOrArray, $propertyPath, $value)
             }
         } catch (\TypeError $e) {
             self::throwInvalidArgumentException($e->getMessage(), $e->getTrace(), 0);
+
+            // It wasn't thrown in this class so rethrow it
+            throw $e;
```

Comment:
the other place is error handler callback for PHP 5, there is no exception to rethrow, and errors are passed correctly to the original handler ([here](https://github.com/tsufeki/symfony/blob/45b961de2eec3faaebfaea9a1ebde32ec1488efe/src/Symfony/Component/PropertyAccess/PropertyAccessor.php#L261))